tests/integration/test_compute_flow.py

```diff
@@ -6,211 +6,153 @@
-    # Give the consumer some datatokens so they can order the service
-    try:
-        dt = DataToken(ocean_instance.web3, ddo.data_token_address)
-        tx_id = dt.transfer(consumer_wallet.address, to_wei(10), publisher_wallet)
-        dt.verify_transfer_tx(tx_id, publisher_wallet.address, consumer_wallet.address)
-    except (AssertionError, Exception) as e:
-        print(e)
-        raise
-
-    # Order compute service from the dataset asset
-    order_requirements = ocean_instance.assets.order(
-        ddo.did, consumer_wallet.address, service_type=service_type
+    # Mint 10 datatokens to the consumer
+    service = asset.get_service(service_type)
+    erc20_token = ERC20Token(ocean_instance.web3, asset.data_token)
+    _ = erc20_token.mint(consumer_wallet.address, to_wei(10), publisher_wallet)
+
+    # Initialize the service to get provider fees
+    initialize_response = DataServiceProvider.initialize(
+        ddo=asset.did,
+        service_id=asset.id,
+        consumer_address=consumer_wallet.address,
+        service_endpoint=DataServiceProvider.build_initialize_endpoint(
+            ocean_instance.config.provider_url
+        )[1],
+    ).json()
+
+    # Order the service
+    order_tx_id = erc20_token.start_order(
+        consumer=consumer_wallet.address,
+        service_id=int(service.id),
+        provider_fees=initialize_response["providerFee"],
+        from_wallet=consumer_wallet,
     )
 
-    # Start the order on-chain using the `order` requirements from previous step
-    service = ddo.get_service(service_type)
-    consumer = service.get_c2d_address()
-    if service_type == ServiceTypes.ASSET_ACCESS and order_requirements.computeAddress:
-        consumer = order_requirements.computeAddress
-
-    _order_tx_id = ocean_instance.assets.pay_for_service(
-        ocean_instance.web3,
-        order_requirements.amount,
-        order_requirements.data_token_address,
-        ddo.did,
-        service.index,
-        ZERO_ADDRESS,
-        consumer_wallet,
-        consumer,
-    )
-    return _order_tx_id, order_requirements, service
+    return order_tx_id, service
 
 
 def run_compute_test(
     ocean_instance,
     publisher_wallet,
     consumer_wallet,
-    input_ddos,
-    algo_ddo=None,
-    algo_meta=None,
+    asset_ddo,
+    algo_ddo,
+    additional_asset_ddos=[],
     expect_failure=False,
     expect_failure_message=None,
     userdata=None,
     with_result=False,
 ):
     """Helper function to bootstrap compute job creation and status checking."""
-    compute_ddo = input_ddos[0]
-    did = compute_ddo.did
-    order_tx_id, _, service = process_order(
+    # Order asset compute service
+    order_tx_id, service = process_order(
         ocean_instance,
         publisher_wallet,
         consumer_wallet,
-        compute_ddo,
+        asset_ddo,
         ServiceTypes.CLOUD_COMPUTE,
     )
-    compute_inputs = [ComputeInput(did, order_tx_id, service.index, userdata=userdata)]
```

Comment:
please reinstate this as plural, we need to be able to process multiple inputs.

Reply:
I intend to do this once the basic compute flow is working.